### ISSUE TITLE
Handle non-initialized image embeddings better in the plugins

### ISSUE THREAD

**Issue by constantinpape** [computational-cell-analytics/micro-sam#332]
Now that we have the embedding widget integrated in the napari plugin, it is possible to click a button that requires the model and image embeddings without having initialized them.  
This fails with a cryptic error message:
```
File ~/Work/my_projects/micro-sam/micro_sam/sam_annotator/_widgets.py:229, in segment_widget(viewer=Viewer(camera=Camera(center=(0.0, 255.5, 255.5),....<locals>._clear_annotations at 0x7f7471efc4a0>}), box_extension=0.1)
    228 image_embeddings = AnnotatorState().image_embeddings
--> 229 if image_embeddings["original_size"] is None:  # tiled prediction
        image_embeddings = None
    230     seg = vutil.prompt_segmentation(
    231         predictor, points, labels, boxes, masks, shape, image_embeddings=image_embeddings,
    232         multiple_box_prompts=True, box_extension=box_extension,
    233     )

TypeError: 'NoneType' object is not subscriptable
```
It would be good to throw a more meaningful error message instead.

We could also indicate visually in the embedding plugin if embeddings are initialized or not.
This could be done via a checkbox `is_initialized` or so  that is not writable from the plugin and just shows if the embeddings are computed or not. (Not sure how to do this exactly and if this is really a good idea.)

**Comment by constantinpape** [comment]
In the long run it would be good to handle this using validation functions in magicgui: https://github.com/pyapp-kit/magicgui/issues/634.
This will probably take a while before being implemented, so we should find an intermediate solution.

**Comment by constantinpape** [comment]
Partially related to this: storing the embedding widgets in the state to have access to it; update the values in the widget to match what is passed in an annotator function, see #483

**Comment by constantinpape** [comment]
This should also encompass handling image embeddings that are wrongly initialized (e.g. for a different image). See also #266

**Comment by constantinpape** [comment]
This is ready to be addressed now that #486 is merged. Relevant locations in the widget are
https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/sam_annotator/_widgets.py#L617 and https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/sam_annotator/_widgets.py#L643.

**Comment by constantinpape** [comment]
Implemented in #498
